vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -61,6 +70,17 @@ def auth_user_pwd(type = nil)
     [cred.userid, cred.password]
   end
 
+  def auth_user_keypair(type = nil)
+    cred =  if self.kind_of?(HostOpenstackInfra)
+              # HostOpenstackInfra is using auth key set on ext_management_system level, not individual hosts
+              self.try(:ext_management_system).try(:authentication_best_fit, type)
```

Comment:
@Fryguy moved to HostOpenstackInfra in dependent PR, till we figure out the general auth system. Looks much nicer, thank you